pytorch_lightning/trainer/training_loop.py

```diff
@@ -206,6 +206,11 @@ def check_checkpoint_callback(self, should_save, is_last=False):
             [c.on_validation_end(self.trainer, model) for c in checkpoint_callbacks]
 
     def on_train_epoch_start(self, epoch):
+
+        # reset train dataloader
+        if self.trainer.reload_dataloaders_every_epoch:
+            self.trainer.reset_train_dataloader(model)
+
```

Comment:
need to swap the order to after model / current_epoch are properly initialized :)